### PR TITLE
Create README file for mutex demo

### PR DESCRIPTION
**Pull request by OscarShiang**
Add description of the mutex demo.

### CODE REVIEW

File: demos/mutexes/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+If the mutex has not been locked, then the thread can get the lock. Thus the function returns `true`.
+
+If a thread tries to lock a mutex that has been locked by another thread, the operation fails and the function returns `false` as a result.
+
+To prevent the thread that failed to lock from accessing shared resources, we need to use a loop to keep it yielding until it can acquire the lock.
+
+## `unlock_mutex`
+
+When a thread finishes its operation to the shared variables protected by the mutex, we use this function to release the lock.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "operations on the shared variables"

File: demos/mutexes/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+After the mutex is unlocked, other threads can acquire the mutex and enter the protected section again.
+
+## Walkthrough
+
+In the beginning, we initialize two threads to do `thread_worker` routine. The routine is to copy a given string to the global array called `buf`.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "to do the `thread worker` routine"
"The routine copies a given string"

File: demos/mutexes/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Since thread 1 has been created earlier, when `setup()` ends, the scheduler chooses it to yield. After entering `thread_worker()` function, thread 1 acquires the lock and starts to copy `"dog"` into `buf`.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "chooses it to run first."

File: demos/mutexes/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Since thread 1 locked the mutex before, thread 2 can not enter the section but need to yield and wait.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "but must yield and wait"

File: demos/mutexes/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Until thread 1 ends its copying, it releases the mutex. Then thread 2 can enter the section and copy `"cat"` to `buf`.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "Once thread 1 ends its copying, it prints `buf` and releases the mutex."

"Then thread 2 can acquire the lock"

File: demos/mutexes/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Finally, when thread 2 prints out its result of copying, the demo ends.
```
Comment (DavidSpickett) [inline demos/mutexes/README.md]: "Finally thread 2 prints the result of its copying, and the demo ends.

**Review by DavidSpickett — CHANGES_REQUESTED** [review]

**Comment by OscarShiang** [comment]
I have fixed in the new commit.

**Review by DavidSpickett — APPROVED** [review]
LGTM, thanks!

**Pull request merged**